Workspace/src/xbox_controller/src/teleop_node.cpp

```diff
@@ -42,18 +61,34 @@ class StateSelect {
   ros::Publisher drive_pub;
   ros::Publisher science_pub;
   ros::Publisher arm_pub;
-  enum topics { DRIVE, ARM, SCIENCE };
+  ros::Publisher autonomy_pub;
+  enum topics { DRIVE, ARM, SCIENCE, AUTONOMY };
   int i; // Variable thats holds the current state index
   bool pressed;
+  sensor_msgs::Joy empty_joy_msg; // empty joy message needed when switching states
 };
 
 // Callback function: Called everytime there is an update on the joy-stick
 void StateSelect::callback(const sensor_msgs::Joy::ConstPtr &joy) {
 
   // logic to cycle through different states
   if (joy->buttons[8] == 1 && !pressed) {
+    // Send empty message
+    if (i == DRIVE)
+      drive_pub.publish(empty_joy_msg);
+
+    else if (i == ARM)
+      arm_pub.publish(empty_joy_msg);
+
+
+    else if (i == SCIENCE)
+      science_pub.publish(empty_joy_msg);
+
+    else if (i == AUTONOMY)
+      autonomy_pub.publish(empty_joy_msg);
+
     i++;
-    if (i >= 3)
+    if (i >= 4)

```

Comment:
This is fine but look into using modulo for this in the future

Reply:
This `4` right here

Reply:
It just makes the code a bit cleaner since if you add a topic to the enum, the NUM_TOPICS value will automatically increment